gateway-service-impl/src/main/java/org/hypertrace/gateway/service/common/datafetcher/QueryServiceEntityFetcher.java

```diff
@@ -740,6 +744,30 @@ private QueryRequest buildTimeSeriesQueryRequest(
                             entityIdAttribute, Operator.NEQ, QUERY_SERVICE_NULL))
                 .map(Filter.Builder::build)
                 .collect(Collectors.toList()));
+
+    // Convert the existing filter into query service filter.
+    Filter queryFilter = convertToQueryFilter(entitiesRequest.getFilter()).build();
+    if (!Filter.getDefaultInstance().equals(queryFilter)) {
+      filterBuilder.addChildFilter(queryFilter);
+    }
+
+    // Time range is a mandatory filter for query service, hence add it if it's not already present.
+    if (!hasTimeRangeFilter(queryFilter, context.getTimestampAttributeId())) {
```

Comment:
This is the actual bug fix, where if the timestamp filter isn't already present, we'll explicitly add it here.

Reply:
Thanks for pointing out. I wonder if there should be a case showing this bug in QueryServiceEntityFetcherTests.java?

Reply:
Good ask. There was a test but the assertion wasn't verifying the complete query service request. Added that now at https://github.com/hypertrace/gateway-service/pull/46/files#diff-8fa94e86f08f72f2fe9bd4fbe8a691758efd65b10513050e7ca568ffbb8f7098R269

Reply:
codecov diff still shows this if branch is not covered.

Reply:
Check now.. it's all green :)

Reply:
ok.